File: day:2/reverseInteger.py
class Solution:
    def reverse(self, x: int) -> int:
        toStr = str(x)
        sign = 1;
        
        if toStr[0] == '-':
            sign = -1
            toStr = toStr[1:]
        
        value = 0
        for i in range(len(toStr)):
            if toStr[i] == '0' or toStr[i] == '-':
                continue
            value += pow(10, i) * int(toStr[i])
        
        value = sign * value;
        if value > (-1 * pow(2, 31)) and value < (pow(2, 31) - 1):
            return value
        else:
            return 0

File: day:2/reverseInteger.java
class Solution {
    public int reverse(int x) {
        int sign = (x < 0)? -1: 1;
        String toStr = String.valueOf(x);
        if(toStr.charAt(0) == '-') {
            toStr = toStr.substring(1);
        }
        long value = 0;
        for(int i = 0; i < toStr.length(); i++) {
            char digit = toStr.charAt(i);
            if(digit == '0' || digit == '-') {
                continue;
            }
            value +=  Math.pow(10, i) * Character.getNumericValue(digit);
            if(value > Integer.MAX_VALUE) {
                return 0;
            }
        }
        value = sign * value;
        return (int)value;
    }
}
